### ISSUE TITLE
BTRFS out of space due to inodes

### ISSUE THREAD

**Issue by golharam** [awslabs/amazon-ebs-autoscale#2]
I enabled this on an AWS node I use to auto-grow my /scratch space as needed.  I recently ran into issues where even though there is free space, BTRFS reports there is no free space.   This has something to do with the way inodes are utilized in BTRFS.   There is a bit of discussion and bug reporting around this: https://btrfs.wiki.kernel.org/index.php/FAQ#Help.21_Btrfs_claims_I.27m_out_of_space.2C_but_it_looks_like_I_should_have_lots_left.21

For production use, BTRFS is not production ready, and I would caution anyone against deploying this as we did.

**Comment by wleepang** [comment]
Sorry to hear that you experienced issues.  Can you provide more details about your use case?  Specifically, how much data had you stored in the expandable file system before encountering the issue?

**Comment by golharam** [comment]
Sure.   I have a Launch Template that AWS Batch uses to run jobs.  The Batch Template mounts a 200GB EBS volume at /scratch with the BTRFS filesystem.   I use this autoscale code to enable it to grow as needed.   Most of the time it works without issue.   Occasionally I would have a job fail.  One of the first things the job does is download files from S3.   After downloading the files, it uncompresses them.  TAR would fail at some point indicating there is no space left on the filesystem.   I've confirmed this is impossible as the tarball is less than 10GB in size.  

I've encountered this in various ways at various times.  On another instance, I started with a 200GB EBS volume, that has since grown to about 12TB in size, spanning over several disks.  I've run into the same issue with disk space issues, when I know there is still free space.

I could never figure out why, but recently I've been able to successfully reproduce this issue and narrowed it down to BTRFS.  Doing some research the other day leads me to believe this is an ongoing issue.
